### DISCUSSION TITLE
Support / upgrade to new hcloud csi 2.x

### DISCUSSION THREAD

**Discussion by ifeulner** [mysticaltech/terraform-hcloud-kube-hetzner#441] [Announcements]
There is a new major version of hcloud CSI that has also breaking changes that must be considered. 
The secret name changed, details see here https://github.com/hetznercloud/csi-driver#from-v1-to-v2

Question is, do we just support 2.x or should we also support 1.x still?

**Comment by otavio** [comment]
I'd vote for support 2.x as an opt-in for now, and after some time move to it. Also, we need to see how users with already installed cluster can do the upgrade.

**Comment by mysticaltech** [comment]
@ifeulner So all installations happening now are actually installing as 2.x the hcloud-csi (I will triple check tonight with a test deploy), since no complaints, it means we already support it. 

Basically, we should just add the link above for folks who want to upgrade their old install. What do you think?

**Comment by mysticaltech** [comment]
My thinking is that we provide a way for folks to deploy an almost maintenance-free cluster, with auto node upgrades, and auto Kube version upgrades, now for components like these, users just have to upgrade their clusters themselves, and we will provide the link in that case. 

I would guess that it would not break the terraform state if they later want to apply to add a new nodepool or increase a nodepool count for instance, as that logic happens in the kustomization part which is supposed to run only on the first terraform apply.

**Comment by ifeulner** [comment]
So just going with the newest version? Then also the secret `hcloud-csi`can be removed, as starting with version 2 the standard `hcloud`secret is used. But this means no backwards-compatibility (and also no upgrade path)

**Comment by mysticaltech** [comment]
@ifeulner I suggest let's leave the old logic in up to our 2.x, since everything is working smooth and the transition has been transparent to us.
